tokio-util/src/codec/length_delimited.rs

```diff
@@ -87,6 +87,7 @@
 //!     .length_field_type::<u16>()
 //!     .length_adjustment(0)   // default value
 //!     .num_skip(0) // Do not strip frame header
+//!     .max_frame_length(255)
 //!     .new_read(io);
```

Comment:
You are adding `.max_frame_length` to a bunch of examples. Do the examples fail without this change?

Reply:
The default value for `max_frame_len` is set to `8 * 1_024 * 1_024` which is appropriate when the length field is `4` bytes long (which is the default). These examples reduce the length field without reducing the `max_frame_len`.

Reply:
If this change causes those examples to panic due to the default value of `max_frame_length`, then I don't think we can make this change. It will be too breaking in that case.

Reply:
What if calling `length_adjustment`, or `length_field_type`, or `length_field_length`, also saturates the `max_frame_len` if it's more than the allowed maximum?

Reply:
Yeah, we probably just document that if `max_frame_length` is greater than what can fit in the length, then the maximum length is actually the maximum that can fit in the length.